### PR TITLE
Inject MogClient instead of building a new instance each time

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Towards #1754

This also speeds up "updateDownloadSize" task as the client is not generated on each map camera move

@gino-m  PTAL?

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/ui/map/gms/mog/MogProviderModule.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+class MogProviderModule {
+  @Provides
+  fun provideMogClient(remoteStorageManager: RemoteStorageManager): MogClient {
+    return MogClient(DEFAULT_MOG_COLLECTION, remoteStorageManager)
+  }
+
+  companion object {
+    private val DEFAULT_MOG_SOURCES = Config.getMogSources(Config.DEFAULT_MOG_TILE_LOCATION)
+    private val DEFAULT_MOG_COLLECTION = MogCollection(DEFAULT_MOG_SOURCES)
+  }
```
Comment (anandwana001) [inline ground/src/main/java/com/google/android/ground/ui/map/gms/mog/MogProviderModule.kt:36]: if both the variables are private, then we can keep them outside the companion object, as these variables are not planned to be accessible from outside. 

WDYT?

Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/map/gms/mog/MogProviderModule.kt]: Makes sense! Done

**Comment by shobhitagarwal1612** [comment]
@gino-m Gentle ping!

**Review by gino-m — APPROVED** [review]
👍 LGTM!

**Pull request merged**